### DISCUSSION TITLE
Pod name from KubernetesPodOperator() comes out without hyphens

### DISCUSSION THREAD

**Discussion by RubcovMindbox** [apache/airflow#20439] [Q&A] [answered]
1. I'd like to have k8s pods with names that contain hyphens. But when I pass such a name (as `task_id`) to `KubernetesPodOperator()` hyphens are removed from the name. Is that expected behaviour or a bug?
2. Why the pod name consist of `dag_id `ant `task_id`. Can we stay only with `task_id` somehow?
Exampe:
DAG code:
![image](https://user-images.githubusercontent.com/87309252/146877426-ce9a76e1-b042-4844-a713-9ccd32f68b7a.png)

Pod spec:
![image](https://user-images.githubusercontent.com/87309252/146877567-3a87858b-52f4-49c6-ad3e-706fc21c1acd.png)

**Comment by potiuk** [comment, marked as answer]
Yes. It is expected. https://github.com/apache/airflow/blob/85bedd03c33a1d6c6339e846558727bf20bc16f7/airflow/kubernetes/kubernetes_helper_functions.py#L29

**Reply by RubcovMindbox** [reply]
Thank you!
